### ISSUE TITLE
Object detection docstrings broken

### ISSUE THREAD

**Issue by jbischof** [keras-team/keras-cv#2018]
In the docstrings for at least [three](https://github.com/search?q=repo%3Akeras-team%2Fkeras-cv%20model.fit(images%2C%20labels)&type=code) OD models/metrics, we create a single example using a tf tensor and Python dictionary. However, they don't seem to work even with `tf.keras` with the following error:

Code:
```python
model.fit(images, labels)
```

```
ValueError: Failed to find data adapter that can handle input: <class 'numpy.ndarray'>, (<class 'dict'> containing {"<class 'str'>"} keys and {'(<class \'list\'> containing values of types {\'(<class \\\'list\\\'> containing values of types {"<class \\\'int\\\'>"})\'})', '(<class \'list\'> containing values of types {\'(<class \\\'list\\\'> containing values of types {\\\'(<class \\\\\\\'list\\\\\\\'> containing values of types {"<class \\\\\\\'int\\\\\\\'>"})\\\'})\'})'} values)
```

See colab [gist](https://colab.research.google.com/gist/jbischof/459b62de475c70c9b731507b0ef5d302/debugging-kerascv-docstrings.ipynb).

**Comment by sampathweb** [comment]
Its not able to find a data adapter in `tf.keras` that can handle the inputs provided.  If we create a TF Dataset, it works fine - 

```
ds = tf.data.Dataset.from_tensor_slices((images, labels)).batch(2)
model.fit(ds)
```
We may need to update the doc string.
This still doesn't work with `keras_core` with TensorFlow backend and has a different error.  Looking into it

**Comment by jbischof** [comment]
@sampathweb if there's a more lightweight approach that works it might be a better fit for example docstrings. However, since most users will be brining tf.data I'm also OK with this.

**Comment by sampathweb** [comment]
The above format of using dataset works with `tf.keras` backend and also with `keras_core` - `jax` and `torch` backend.  Strangely it fails with TensorFlow backend.  Looking more into it.

```
import os
os.environ["KERAS_BACKEND"] = "tensorflow" # Fails
os.environ["KERAS_BACKEND"] = "jax" # Works
os.environ["KERAS_BACKEND"] = "torch" # Works

images = np.ones(shape=(1, 512, 512, 3))
labels = {
    "boxes": [
        [
            [0, 0, 100, 100],
            [100, 100, 200, 200],
            [300, 300, 400, 400],
        ]
    ],
    "classes": [[1, 1, 1]],
}
model = keras_cv.models.RetinaNet.from_preset(
    "retinanet_resnet50_pascalvoc",
    num_classes=20,
    bounding_box_format="xywh",
)

ds = tf.data.Dataset.from_tensor_slices((images, labels)).batch(2)
# Train model
model.compile(
    classification_loss='focal',
    box_loss='smoothl1',
    optimizer=keras.optimizers.SGD(global_clipnorm=10.0),
    jit_compile=False,
)
model.fit(ds)
```

**Comment by sampathweb** [comment]
Addressed in #2119
